### PR TITLE
Feature/v8 load video texture asset

### PR DESCRIPTION
**Pull request by achamas-playco**
Note: This PR also adds a `preload` option to `VideoSource`. Tests included.

### <samp>🤖 Generated by Copilot at 79a2bc9</samp>

This pull request adds support for loading video textures from various sources using the `Assets` class and the `loadVideoTextures` function. It also refactors the `VideoSource` class and its tests to improve the performance and usability of video textures. It removes some unused and deprecated code related to video sources and cross-origin handling.

### CODE REVIEW

File: src/rendering/renderers/shared/texture/sources/VideoSource.ts [hunk 1]
```diff
@@ -85,96 +87,22 @@ export class VideoSource extends TextureSource<VideoResource>
...
-     * @param {boolean} [options.autoPlay=true] - Start playing video immediately
-     * @param {number} [options.updateFPS=0] - How many times a second to update the texture from the video.
-     * Leave at 0 to update at every render.
-     * @param {boolean} [options.crossorigin=true] - Load image using cross origin
-     * @param {boolean} [options.loop=false] - Loops the video
-     * @param {boolean} [options.muted=false] - Mutes the video audio, useful for autoplay
-     * @param {boolean} [options.playsinline=true] - Prevents opening the video on mobile devices
      */
-    public static from(
+    public static async from(
```
Comment (achamas-playco) [inline src/rendering/renderers/shared/texture/sources/VideoSource.ts:91]: Will remove

File: tests/renderering/textures/VideoSource.test.ts [hunk 2]
```diff
@@ -48,50 +48,43 @@ describe('VideoSource', () =>
...
         expect(sourceElement.type).toEqual('video/mp4');
-
-        source.destroy();
     });
 
     it('should respect the updateFPS settings property and getter / setter', async () =>
     {
-        const { source } = setup({ updateFPS: 30 });
+        // clear the cache since we're creating new properties for the existing source
+        Assets.reset();
```
Comment (achamas-playco) [inline tests/renderering/textures/VideoSource.test.ts:70]: will add `onBefore`

File: src/assets/loader/parsers/textures/loadVideoTextures.ts [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+        return isValidDataUrl || isValidExtension;
+    },
+
+    async load(url: string, asset: ResolvedAsset<TextureSourceOptions>, loader: Loader): Promise<Texture>
+    {
+        // --- Merge default and provided options ---
+        const options: VideoSourceOptions = {
+            ...VideoSource.defaultOptions,
+            resolution: asset.data?.resolution || getResolutionOfUrl(url), // Assume getResolutionOfUrl is globally available
```
Comment (Zyie) [inline src/assets/loader/parsers/textures/loadVideoTextures.ts:110]: let's also add `alphaMode: await detectVideoAlphaMode()` here

**Review by Zyie — APPROVED** [review]

**Comment by xeoshow** [comment]
Just would like to know if this feature is also added to v7 latest version? 
If yes, From which version of v7 is supported? Thanks a lot.

**Pull request merged**